### PR TITLE
Create the `web-sys` crate mechanically from WebIDL

### PR DESCRIPTION
**Pull request by fitzgen**
I vendored all the WebIDL from mozilla-central and then used an apache-style available/enabled symlinking thing to choose which ones we actually generate bindings for. Right now the only thing we actually generate bindings for is `Event`.

See the changes to the guide for details on the expected workflow for adding support for more Web APIs.

This is still WIP because I'm running into issues getting the project builder working for different crates. The way it wants to link `package.json` and friends is a bit funky. Nothing too hard to fix, but I am not going to come back to this for a couple days and figured folks might want to provide some early feedback.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -58,6 +58,17 @@ matrix:
...
           done
 
+    # The `web-sys` crate's tests pass on nightly.
+    - rust: nightly
+      env: JOB=test-web-sys
+      before_install: *INSTALL_NODE_VIA_NVM
+      install:
+        - npm install
+      script:
+        - (cd crates/web-sys && cargo test)
```
Comment (alexcrichton) [inline .travis.yml]: I'm personally also a fan of `cargo test --manifest-path crates/web-sys/Cargo.toml`

File: guide/src/web-sys/overview.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+These are all the different WebIDL definitions we intend to support. At the time
+of writing, we don't even attempt to generate bindings for most of these.
+
+### `webidls/enabled/*.webidl`
+
+These are the WebIDL interfaces that we will actually generate bindings for (or
+at least bindings for *some* of the things defined in these files).
+
+These are all symlinks into the `webidls/available` directory.
```
Comment (alexcrichton) [inline guide/src/web-sys/overview.md]: Given that windows doesn't support symlinks very well, could we instead perhaps move files from `available` to ` enabled`?

**Comment by fitzgen** [comment]
Looks like appveyor headless testing is failing with this now:

```
errors:
    exception: __wbg_f_values_values_Array_target is undefined
    stack: __wbg_f_values_values_Array@http://localhost:8080/1.bundle.js:695:5
    __wbg_f_values_values_Array@http://localhost:8080/bundle.js:59:23
    wasm_bindgen::js::Array::values::h31b91cbdcb83beef@http://localhost:8080/bundle.js:2676:1
    test80::get_values::h95694f03c9da97e7@http://localhost:8080/bundle.js:1544:1
    get_values@http://localhost:8080/bundle.js:1740:1
    get_values@http://localhost:8080/1.bundle.js:312:27
    test@http://localhost:8080/0.bundle.js:22:36
    run@http://localhost:8080/bundle.js:9489:23
    ./run.js/
```

**Comment by fitzgen** [comment]
> Looks like appveyor headless testing is failing with this now:

I think this is failing on master as well.

**Comment by alexcrichton** [comment]
Bah that's probably myself botching a merge, sorry about that! Looks like AppVeyor is running Firefox 59 but Array.prototype.values was added in Firefox 60, which'd explain the failing test.

File: src/js.rs [hunk 3]
```diff
@@ -443,6 +443,16 @@ extern "C" {
     pub fn to_string(this: &Function) -> JsString;
 }
 
+impl JsValue {
+    pub fn as_function(&self) -> Option<Function> {
```
Comment (alexcrichton) [inline src/js.rs]: Should this perhaps be `&Function` via a transmute? (same w/ `as_object` below)

Comment (fitzgen) [reply src/js.rs]: On it.

**Comment by fitzgen** [comment]
Ok, finally got CI green! Going to merge!

**Pull request merged**